API doc cleanup for decorators

jbogarthyde opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

## What is the new behavior?
Makes descriptions parallel, cleans up styles and formats to be consistent.
Mostly for decorators, but some other API issues that were reported.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

jbogarthyde committed:
docs: api doc cleanup

mary-poppins commented:
You can preview 4fb4ff3 at https://pr31377-4fb4ff3.ngbuilds.io/.

mary-poppins commented:
You can preview f2ed881 at https://pr31377-f2ed881.ngbuilds.io/.

mary-poppins commented:
You can preview 971dc01 at https://pr31377-971dc01.ngbuilds.io/.

mary-poppins commented:
You can preview f76f08b at https://pr31377-f76f08b.ngbuilds.io/.

IgorMinar commented on packages/core/src/di/interface/provider.ts:
```diff
@@ -123,37 +117,35 @@ export interface StaticClassProvider extends StaticClassSansProvider {
   /**
-   * A list of `token`s which need to be resolved by the injector. The list of values is then
-   * used as arguments to the `useClass` constructor.
+   * A list of `token`s  to be resolved by the injector.
```
```suggestion
   * A list of `token`s to be resolved by the injector.
```

IgorMinar commented on packages/core/src/di/interface/provider.ts:
```diff
@@ -162,62 +154,56 @@ export interface ConstructorProvider extends ConstructorSansProvider {
  * Configures the `Injector` to return a value of another `useExisting` token.
  *
- * For more details, see the ["Dependency Injection Guide"](guide/dependency-injection).
+ * @see ["Dependency Injection Guide"](guide/dependency-injection).
```
```suggestion
 * @see `ExistingProvider`
 * @see ["Dependency Injection Guide"](guide/dependency-injection).
```

IgorMinar commented on packages/core/src/di/interface/provider.ts:
```diff
@@ -162,62 +154,56 @@ export interface ConstructorProvider extends ConstructorSansProvider {
- * ### Example
- *
- * {@example core/di/ts/provider_spec.ts region='ExistingSansProvider'}
+ * {@example core/di/ts/provider_spec.ts region='ExistingSansProvider' linenums="false"}
```
does this region actually exist? I don't see it defined in the provider_spec.ts file.

I just noticed that `ConstructorProvider` is not exported from @angular/core at all (this makes it inaccessible to developers for direct interaction but allows type inference and typechecking to work just fine - meaning that developers would usually not complain about the lack of exporting but the docs for this symbol are missing on angular.io)

can you please fix this my exporting the symbol from packages/core/src/di/index.t and update tools/public_api_guard/core/core.d.ts to reflect the adition?

jbogarthyde replied:
I can update the index, but I don't have access to the golden file.  I've asked George to do it.

gkalpak replied:
I foudn out there were a few more missing, so added them all: https://github.com/angular/angular/pull/31377/commits/6b5bf085d6b9b00a266bb1a9982df302e3f49867

IgorMinar commented on packages/core/src/linker/component_factory_resolver.ts:
```diff
@@ -34,10 +34,19 @@ class _NullComponentFactoryResolver implements ComponentFactoryResolver {
 }
 
 /**
+ * Stores factory classes that have been defined to create components of specific types.
```
```suggestion
 * A simple registry that maps `Components` to generated `ComponentFactory` classes that can be used to create instances of components.
```

IgorMinar reviewed (approved):
great improvements! thank you

I left a few comments / suggestions.

devversion commented on packages/core/src/metadata/di.ts:
```diff
@@ -188,33 +189,34 @@ export const ContentChildren: ContentChildrenDecorator = makePropDecorator(
+   * * **read** - True to read a different token from the queried element.
+   * * **static** - True to resolve query results before change detection runs.
+   *
+   * When `static` is false or not provided, uses query results to determine the timing of query
```
This seems to be wrong. If the `static` flag is set to `false`, the query timing is not determined automatically.

devversion replied:
This seems wrong to me. If the `static` flag is set to `false`, the query timing is not resolved automatically. The results will never be available before change detection ran (at least based on https://github.com/angular/angular/pull/28810)

mary-poppins commented:
You can preview 01ddfc4 at https://pr31377-01ddfc4.ngbuilds.io/.

mary-poppins commented:
You can preview 967f440 at https://pr31377-967f440.ngbuilds.io/.

mary-poppins commented:
You can preview c25a0f9 at https://pr31377-c25a0f9.ngbuilds.io/.

mary-poppins commented:
You can preview c3dd1ff at https://pr31377-c3dd1ff.ngbuilds.io/.

gkalpak committed:
fix(core): export provider interfaces that are part of the public API types

Some of the provider interfaces that the [Provider][1] and
[StaticProvider][2] types comprise were not exported from
[@angular/core][3]. As a result, the docs for these symbols did not
appear on angular.io (even though both `Provider` and `StaticProvider`
are part of the public API. (See, also,
https://github.com/angular/angular/pull/31377#discussion_r299254408.)

This commit fixes it by exporting all necessary provider interfaces.

[1]: https://github.com/angular/angular/blob/9e34670b2/packages/core/src/di/interface/provider.ts#L365-L366
[2]: https://github.com/angular/angular/blob/9e34670b2/packages/core/src/di/interface/provider.ts#L283-L284
[3]: https://github.com/angular/angular/blob/9e34670b2/packages/core/src/di/index.ts#L23

mary-poppins commented:
You can preview d457932 at https://pr31377-d457932.ngbuilds.io/.

mary-poppins commented:
You can preview f1e57b8 at https://pr31377-f1e57b8.ngbuilds.io/.

jbogarthyde commented:
Please fix the CLA and re-run codefresh.

mary-poppins commented:
You can preview 4a76e3c at https://pr31377-4a76e3c.ngbuilds.io/.

mary-poppins commented:
You can preview 3e38ced at https://pr31377-3e38ced.ngbuilds.io/.

mary-poppins commented:
You can preview e6a1678 at https://pr31377-e6a1678.ngbuilds.io/.

mary-poppins commented:
You can preview 742b170 at https://pr31377-742b170.ngbuilds.io/.
